core/src/main/java/ai/timefold/solver/core/impl/heuristic/selector/move/generic/list/ListSwapMoveSelector.java

```diff
@@ -41,41 +43,36 @@ public void solvingStarted(SolverScope<Solution_> solverScope) {
         var listVariableDescriptor = (ListVariableDescriptor<Solution_>) leftValueSelector.getVariableDescriptor();
         var supplyManager = solverScope.getScoreDirector().getSupplyManager();
         listVariableStateSupply = supplyManager.demand(listVariableDescriptor.getStateDemand());
-        movableLeftValueSelector = filterPinnedListPlanningVariableValuesWithIndex(leftValueSelector, listVariableStateSupply);
-        movableRightValueSelector =
-                filterPinnedListPlanningVariableValuesWithIndex(rightValueSelector, listVariableStateSupply);
     }
 
     @Override
     public void solvingEnded(SolverScope<Solution_> solverScope) {
         super.solvingEnded(solverScope);
         listVariableStateSupply = null;
-        movableLeftValueSelector = null;
```

Comment:
I liked the simplification made with the "effective" selectors, and I noted a pattern where we recreate the selectors when the solver starts. You might have already considered this before making the changes, but would it be possible for the selectors to be inconsistent if we don't recreate them when the solver starts?

Reply:
Where have you noticed me recreating selectors during solvingStarted()?
If there is still a place like that, it's a bug and it needs to be removed.

Reply:
I did not check any other selector outside this PR. The patterns I noted are the ones I shared before.

Reply:
I checked all the selectors but couldn't find any that recreates the selectors at the `solvingStarted` method. Regardless, the question still remains: Could not recreating the selectors lead to an inconsistent state?

Reply:
No. Selectors don't really have state, until solvingStarted() is called on them. The way we recreate selectors is by wrapping them - the parent always delegates the lifecycle calls to the child. Since the recreation now happens in the selector constructor, none of the selectors could have had any lifecycle methods called on them yet. Therefore this is wrapping selectors without state with other selectors without state. This is safe, unlike what was there previously.